### ISSUE TITLE
java.lang.IllegalStateException Can not perform this action after onSaveInstanceState

### ISSUE THREAD

**Issue by gabrielbull** [grahammendick/navigation#353]
Another issue that gets reported a lot in my app. I have not yet found a way to reproduce.

```
java.lang.IllegalStateException Can not perform this action after onSaveInstanceState 
    FragmentManagerImpl.java:1536 androidx.fragment.app.FragmentManagerImpl.checkStateLoss
    FragmentManagerImpl.java:1558 androidx.fragment.app.FragmentManagerImpl.enqueueAction
    BackStackRecord.java:317 androidx.fragment.app.BackStackRecord.commitInternal
    BackStackRecord.java:282 androidx.fragment.app.BackStackRecord.commit
    FragmentNavigator.java:64 com.navigation.reactnative.FragmentNavigator.navigate
    NavigationStackView.java:62 com.navigation.reactnative.NavigationStackView.onAfterUpdateTransaction
    NavigationStackView.java:74 com.navigation.reactnative.NavigationStackView.onAttachedToWindow
    View.java:18451 android.view.View.dispatchAttachedToWindow
    ViewGroup.java:3462 android.view.ViewGroup.dispatchAttachedToWindow
    ViewGroup.java:5148 android.view.ViewGroup.addViewInner
    ViewGroup.java:4930 android.view.ViewGroup.addView
    ReactViewGroup.java:453 com.facebook.react.views.view.ReactViewGroup.addView
    ViewGroup.java:4870 android.view.ViewGroup.addView
    ViewGroupManager.java:35 com.facebook.react.uimanager.ViewGroupManager.addView
    NativeViewHierarchyManager.java:512 com.facebook.react.uimanager.NativeViewHierarchyManager.manageChildren
    UIViewOperationQueue.java:205 com.facebook.react.uimanager.UIViewOperationQueue$ManageChildrenOperation.execute
    UIViewOperationQueue.java:779 com.facebook.react.uimanager.UIViewOperationQueue$1.run
    UIViewOperationQueue.java:888 com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches
    UIViewOperationQueue.java:42 com.facebook.react.uimanager.UIViewOperationQueue.access$2200
    UIViewOperationQueue.java:846 com.facebook.react.uimanager.UIViewOperationQueue$2.runGuarded
    GuardedRunnable.java:29 com.facebook.react.bridge.GuardedRunnable.run
    Handler.java:873 android.os.Handler.handleCallback
    Handler.java:99 android.os.Handler.dispatchMessage
    Looper.java:193 android.os.Looper.loop
    ActivityThread.java:6715 android.app.ActivityThread.main
    Method.java:-2 java.lang.reflect.Method.invoke
    RuntimeInit.java:493 com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run
    ZygoteInit.java:911 com.android.internal.os.ZygoteInit.main
```

**Comment by gabrielbull** [comment]
If you need paid support or help to tackle these issues, I will gladly pay or help. You can always drop me a message by email to talk privately: me@gabrielbull.com

**Comment by grahammendick** [comment]
Thanks, that's kind of you but there's no need. I'm glad you're raising issues and working with me to fix them

**Comment by grahammendick** [comment]
I've recreated it. Call `stateNavigator.navigate` in the JavaScript and, before it executes on the Java side, press the Android Home button. This backgrounds the app and calls `onSaveInstanceState`. Then the navigate call executes on the Java side. This throws the exception because  can't commit a fragment change after `onSaveInstanceState`
. 
I'm not sure of the fix yet, but recreating it is half the battle

**Comment by grahammendick** [comment]
I had another look at this and they way I recreated it doesn't give the same stack trace. Yours is happening when the stack first loads. Do you ever create a new StateNavigator when the app is running (inside the App component)?

**Comment by gabrielbull** [comment]
Well, the more issue we can fix the better right :-). Theoretically, I am never creating a new StateNavigator. But I've noticed issues when using [react-native-splash-screen ](https://github.com/crazycodeboy/react-native-splash-screen) and that may be what's happening here. I'll try to remove it and test if it's the cause of some of the issues.

**Comment by grahammendick** [comment]
Try pressing the Android Home button while the splash screen is visible

**Comment by grahammendick** [comment]
I've created your stack trace now by pressing the Android Home button when the app is first loading. I pressed it early, before the first screen is displayed.

**Comment by grahammendick** [comment]
@gabrielbull I've put a fix together in #356. Could you give it a thorough test like you did last time, please?

It needs a proper test because the Android documentation says that in [onResume](https://developer.android.com/reference/android/support/v4/app/FragmentActivity.html#onresume)
> in some cases the previous state may still be saved not allowing fragment transactions that modify the state.

Unfortunately I can't access the recommended `onPostResume` because React Native doesn't make it available. However, in my testing so far `onResume` has worked out fine so I'm hoping we're not affected by the edge cases.

**Comment by grahammendick** [comment]
Don't worry, I tested lots of different scenarios and it works well
